Author: Alberto Luaces <aluaces@udc.es>
Description: Split the .pc file into smaller chunks, one per OSG library. This way, programs linked to OSG won't have unneeded references as well.
--- a/OpenSceneGraph/CMakeLists.txt
+++ b/OpenSceneGraph/CMakeLists.txt
@@ -777,14 +777,36 @@
     ENDIF(BUILD_OSG_PACKAGES)
 ENDIF(CMAKE_CPACK_COMMAND)
 
-# Generate pkg-config configuration file
+# Generate pkg-config configuration files
 
-CONFIGURE_FILE(${PROJECT_SOURCE_DIR}/packaging/pkgconfig/openscenegraph.pc.in
-        ${PROJECT_BINARY_DIR}/packaging/pkgconfig/openscenegraph.pc
+SET(PKGCONFIG_FILES
+  openscenegraph
+  openscenegraph-osg
+  openscenegraph-osgDB
+  openscenegraph-osgFX
+  openscenegraph-osgGA
+  openscenegraph-osgParticle
+  openscenegraph-osgSim
+  openscenegraph-osgText
+  openscenegraph-osgUtil
+  openscenegraph-osgTerrain
+  openscenegraph-osgManipulator
+  openscenegraph-osgViewer
+  openscenegraph-osgWidget
+  openscenegraph-osgShadow
+  openscenegraph-osgAnimation
+  openscenegraph-osgVolume
+  openscenegraph-osgIntrospection
+)
+
+FOREACH(PKGCONFIG_FILE ${PKGCONFIG_FILES})
+  CONFIGURE_FILE(${PROJECT_SOURCE_DIR}/packaging/pkgconfig/${PKGCONFIG_FILE}.pc.in
+    ${PROJECT_BINARY_DIR}/packaging/pkgconfig/${PKGCONFIG_FILE}.pc
     @ONLY
     )
+  INSTALL(FILES ${PROJECT_BINARY_DIR}/packaging/pkgconfig/${PKGCONFIG_FILE}.pc DESTINATION lib${LIB_POSTFIX}/pkgconfig COMPONENT libopenscenegraph-dev)
+ENDFOREACH(PKGCONFIG_FILE)
 
-INSTALL(FILES ${PROJECT_BINARY_DIR}/packaging/pkgconfig/openscenegraph.pc DESTINATION lib${LIB_POSTFIX}/pkgconfig COMPONENT libopenscenegraph-dev)
 
 # Run this as late as possible so users can easier spot the message
 IF (${CMAKE_INSTALL_PREFIX} STREQUAL "/usr/local")
--- /dev/null
+++ b/OpenSceneGraph/packaging/pkgconfig/openscenegraph-osg.pc.in
@@ -0,0 +1,14 @@
+# pkg-config source file
+
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=${exec_prefix}/lib@LIB_POSTFIX@
+includedir=${prefix}/include
+
+Name: openscenegraph-osg
+Description: 3D scenegraph
+Version: @OPENSCENEGRAPH_VERSION@
+Requires: openthreads
+Conflicts:
+Libs: -L${libdir} -losg
+Cflags: -I${includedir}
--- /dev/null
+++ b/OpenSceneGraph/packaging/pkgconfig/openscenegraph-osgAnimation.pc.in
@@ -0,0 +1,14 @@
+# pkg-config source file
+
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=${exec_prefix}/lib@LIB_POSTFIX@
+includedir=${prefix}/include
+
+Name: openscenegraph-osgAnimation
+Description: Skinning and morphing library for Openscenegraph
+Version: @OPENSCENEGRAPH_VERSION@
+Requires: openscenegraph-osg openthreads
+Conflicts:
+Libs: -L${libdir} -losgAnimation
+Cflags: -I${includedir}
--- /dev/null
+++ b/OpenSceneGraph/packaging/pkgconfig/openscenegraph-osgDB.pc.in
@@ -0,0 +1,14 @@
+# pkg-config source file
+
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=${exec_prefix}/lib@LIB_POSTFIX@
+includedir=${prefix}/include
+
+Name: openscenegraph-osgDB
+Description: File format reading library for Openscenegraph
+Version: @OPENSCENEGRAPH_VERSION@
+Requires: openscenegraph-osg openthreads
+Conflicts:
+Libs: -L${libdir} -losgDB
+Cflags: -I${includedir}
--- /dev/null
+++ b/OpenSceneGraph/packaging/pkgconfig/openscenegraph-osgFX.pc.in
@@ -0,0 +1,14 @@
+# pkg-config source file
+
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=${exec_prefix}/lib@LIB_POSTFIX@
+includedir=${prefix}/include
+
+Name: openscenegraph-osgFX
+Description: Special effects library for Openscenegraph
+Version: @OPENSCENEGRAPH_VERSION@
+Requires: openscenegraph-osgDB openscenegraph-osgUtil openscenegraph-osg openthreads
+Conflicts:
+Libs: -L${libdir} -losgFX
+Cflags: -I${includedir}
--- /dev/null
+++ b/OpenSceneGraph/packaging/pkgconfig/openscenegraph-osgGA.pc.in
@@ -0,0 +1,14 @@
+# pkg-config source file
+
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=${exec_prefix}/lib@LIB_POSTFIX@
+includedir=${prefix}/include
+
+Name: openscenegraph-osgGA
+Description: GUI event library for Openscenegraph
+Version: @OPENSCENEGRAPH_VERSION@
+Requires: openscenegraph-osgDB openscenegraph-osgUtil openscenegraph-osg openthreads
+Conflicts:
+Libs: -L${libdir} -losgGA
+Cflags: -I${includedir}
--- /dev/null
+++ b/OpenSceneGraph/packaging/pkgconfig/openscenegraph-osgIntrospection.pc.in
@@ -0,0 +1,14 @@
+# pkg-config source file
+
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=${exec_prefix}/lib@LIB_POSTFIX@
+includedir=${prefix}/include
+
+Name: openscenegraph-osgIntrospection
+Description: Introspection library for Openscenegraph
+Version: @OPENSCENEGRAPH_VERSION@
+Requires: openscenegraph-osg openthreads
+Conflicts:
+Libs: -L${libdir} -losgIntrospection
+Cflags: -I${includedir}
--- /dev/null
+++ b/OpenSceneGraph/packaging/pkgconfig/openscenegraph-osgManipulator.pc.in
@@ -0,0 +1,14 @@
+# pkg-config source file
+
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=${exec_prefix}/lib@LIB_POSTFIX@
+includedir=${prefix}/include
+
+Name: openscenegraph-osgManipulator
+Description: Manipulator library for Openscenegraph
+Version: @OPENSCENEGRAPH_VERSION@
+Requires: openscenegraph-osgUtil openscenegraph-osgGA openscenegraph-osg openthreads
+Conflicts:
+Libs: -L${libdir} -losgManipulator
+Cflags: -I${includedir}
--- /dev/null
+++ b/OpenSceneGraph/packaging/pkgconfig/openscenegraph-osgParticle.pc.in
@@ -0,0 +1,14 @@
+# pkg-config source file
+
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=${exec_prefix}/lib@LIB_POSTFIX@
+includedir=${prefix}/include
+
+Name: openscenegraph-osgParticle
+Description: Particle system library for Openscenegraph
+Version: @OPENSCENEGRAPH_VERSION@
+Requires: openscenegraph-osgDB openscenegraph-osgUtil openscenegraph-osg openthreads
+Conflicts:
+Libs: -L${libdir} -losgParticle
+Cflags: -I${includedir}
--- /dev/null
+++ b/OpenSceneGraph/packaging/pkgconfig/openscenegraph-osgShadow.pc.in
@@ -0,0 +1,14 @@
+# pkg-config source file
+
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=${exec_prefix}/lib@LIB_POSTFIX@
+includedir=${prefix}/include
+
+Name: openscenegraph-osgShadow
+Description: Shadow techniques library for Openscenegraph
+Version: @OPENSCENEGRAPH_VERSION@
+Requires: openscenegraph-osgUtil openscenegraph-osgDB openscenegraph-osg openthreads
+Conflicts:
+Libs: -L${libdir} -losgShadow
+Cflags: -I${includedir}
--- /dev/null
+++ b/OpenSceneGraph/packaging/pkgconfig/openscenegraph-osgSim.pc.in
@@ -0,0 +1,14 @@
+# pkg-config source file
+
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=${exec_prefix}/lib@LIB_POSTFIX@
+includedir=${prefix}/include
+
+Name: openscenegraph-osgSim
+Description: Simulator utility library for Openscenegraph
+Version: @OPENSCENEGRAPH_VERSION@
+Requires: openscenegraph-osgText openscenegraph-osgDB openscenegraph-osgUtil openscenegraph-osg openthreads
+Conflicts:
+Libs: -L${libdir} -losgSim
+Cflags: -I${includedir}
--- /dev/null
+++ b/OpenSceneGraph/packaging/pkgconfig/openscenegraph-osgTerrain.pc.in
@@ -0,0 +1,14 @@
+# pkg-config source file
+
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=${exec_prefix}/lib@LIB_POSTFIX@
+includedir=${prefix}/include
+
+Name: openscenegraph-osgTerrain
+Description: Terrain library for Openscenegraph
+Version: @OPENSCENEGRAPH_VERSION@
+Requires: openscenegraph-osgUtil openscenegraph-osgDB openscenegraph-osg openthreads
+Conflicts:
+Libs: -L${libdir} -losgTerrain
+Cflags: -I${includedir}
--- /dev/null
+++ b/OpenSceneGraph/packaging/pkgconfig/openscenegraph-osgText.pc.in
@@ -0,0 +1,14 @@
+# pkg-config source file
+
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=${exec_prefix}/lib@LIB_POSTFIX@
+includedir=${prefix}/include
+
+Name: openscenegraph-osgText
+Description: Text rendering library for Openscenegraph
+Version: @OPENSCENEGRAPH_VERSION@
+Requires: openscenegraph-osgDB openscenegraph-osg openthreads
+Conflicts:
+Libs: -L${libdir} -losgText
+Cflags: -I${includedir}
--- /dev/null
+++ b/OpenSceneGraph/packaging/pkgconfig/openscenegraph-osgUtil.pc.in
@@ -0,0 +1,14 @@
+# pkg-config source file
+
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=${exec_prefix}/lib@LIB_POSTFIX@
+includedir=${prefix}/include
+
+Name: openscenegraph-osgUtil
+Description: Utility library for Openscenegraph
+Version: @OPENSCENEGRAPH_VERSION@
+Requires: openscenegraph-osg openthreads
+Conflicts:
+Libs: -L${libdir} -losgUtil
+Cflags: -I${includedir}
--- /dev/null
+++ b/OpenSceneGraph/packaging/pkgconfig/openscenegraph-osgViewer.pc.in
@@ -0,0 +1,14 @@
+# pkg-config source file
+
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=${exec_prefix}/lib@LIB_POSTFIX@
+includedir=${prefix}/include
+
+Name: openscenegraph-osgViewer
+Description: Windowing system setup, thread scene manager library for Openscenegraph
+Version: @OPENSCENEGRAPH_VERSION@
+Requires: openscenegraph-osgText openscenegraph-osgDB openscenegraph-osgUtil openscenegraph-osgGA openscenegraph-osg openthreads
+Conflicts:
+Libs: -L${libdir} -losgViewer
+Cflags: -I${includedir}
--- /dev/null
+++ b/OpenSceneGraph/packaging/pkgconfig/openscenegraph-osgVolume.pc.in
@@ -0,0 +1,14 @@
+# pkg-config source file
+
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=${exec_prefix}/lib@LIB_POSTFIX@
+includedir=${prefix}/include
+
+Name: openscenegraph-osgVolume
+Description: Volume library for Openscenegraph
+Version: @OPENSCENEGRAPH_VERSION@
+Requires: openscenegraph-osgUtil openscenegraph-osgDB openscenegraph-osgGA openscenegraph-osg openthreads
+Conflicts:
+Libs: -L${libdir} -losgVolume
+Cflags: -I${includedir}
--- /dev/null
+++ b/OpenSceneGraph/packaging/pkgconfig/openscenegraph-osgWidget.pc.in
@@ -0,0 +1,14 @@
+# pkg-config source file
+
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=${exec_prefix}/lib@LIB_POSTFIX@
+includedir=${prefix}/include
+
+Name: openscenegraph-osgWidget
+Description: Windowing toolkit library for Openscenegraph
+Version: @OPENSCENEGRAPH_VERSION@
+Requires: openscenegraph-osgText openscenegraph-osgViewer openscenegraph-osgDB openscenegraph-osgGA openscenegraph-osg openthreads
+Conflicts:
+Libs: -L${libdir} -losgWidget
+Cflags: -I${includedir}
